pkg/ingester/ingester.go

```diff
@@ -1691,6 +1703,18 @@ func (i *Ingester) QueryStream(req *client.QueryRequest, stream client.Ingester_
 	return nil
 }
 
+// Dump context error for diagnosis.
+func dumpContextError(ctx context.Context, err error, start, deadline time.Time, deadlineSet bool, spanlog *spanlogger.SpanLogger) {
+	var timeout string
+	if deadlineSet {
+		timeout = fmt.Sprintf("%.2f seconds", deadline.Sub(start).Seconds())
+	} else {
+		timeout = "not set"
+	}
+	level.Warn(spanlog).Log("msg", "query context error", "cause", context.Cause(ctx), "timeout", timeout,
```

Comment:
The querier legitimately cancel the context once it reaches the quorum (e.g. send request to all ingester zones, cancel outstanding requests once reached successful answers from 2 out of 3 zones). I think this will generate a lot of noise in the logs. I suggest you to to put this debug info only in the tracing span (and not log it as well), if your purpose will be to investigate it through tracing anyway.

Reply:
It's a bit difficult to say before we try in practice (I'm more used to logs than traces), but obtaining this info from traces is probably enough.

Reply:
I've changed to `level.Debug` instead, since that what happens elsewhere in `Ingester.QueryStream`, and it should cause only a trace to be emitted unless debug logs are enabled. That should be sufficient, or would it be too noisy even for debug logs?